com.unity.render-pipelines.core/Runtime/Textures/RTHandleSystem.cs

```diff
@@ -221,19 +226,54 @@ public void SetReferenceSize(int width, int height, MSAASamples msaaSamples, boo
             {
                 Vector2Int maxSize = new Vector2Int(GetMaxWidth(), GetMaxHeight());
                 // Making the final scale in 'drs' space, since the final scale must account for rounding pixel values.
-                var scaledFinalViewport = DynamicResolutionHandler.instance.ApplyScalesOnSize(DynamicResolutionHandler.instance.finalViewport);
-                var scaledMaxSize = DynamicResolutionHandler.instance.ApplyScalesOnSize(maxSize);
-                float xScale = (float)scaledFinalViewport.x / (float)scaledMaxSize.x;
-                float yScale = (float)scaledFinalViewport.y / (float)scaledMaxSize.y;
-                m_RTHandleProperties.rtHandleScale = new Vector4(xScale, yScale, m_RTHandleProperties.rtHandleScale.x, m_RTHandleProperties.rtHandleScale.y);
+                var scales = CalculateScaledRatioAgainstMaxSize(DynamicResolutionHandler.instance.finalViewport);
+                m_RTHandleProperties.rtHandleScale = new Vector4(scales.x, scales.y, m_RTHandleProperties.rtHandleScale.x, m_RTHandleProperties.rtHandleScale.y);
             }
             else
             {
-                Vector2 maxSize = new Vector2(GetMaxWidth(), GetMaxHeight());
-                Vector2 scaleCurrent = m_RTHandleProperties.currentViewportSize / maxSize;
+                var scales = CalculateRatioAgainstMaxSize(m_RTHandleProperties.currentViewportSize);
                 Vector2 scalePrevious = m_RTHandleProperties.previousViewportSize / lastFrameMaxSize;
-                m_RTHandleProperties.rtHandleScale = new Vector4(scaleCurrent.x, scaleCurrent.y, scalePrevious.x, scalePrevious.y);
+                m_RTHandleProperties.rtHandleScale = new Vector4(scales.x, scales.y, scalePrevious.x, scalePrevious.y);
+            }
+
+            SetPostProcessScale(m_RTHandleProperties.currentViewportSize.x, m_RTHandleProperties.currentViewportSize.y);
+        }
+
+        internal void SetPostProcessScale(int width, int height)
+        {
+            Vector2 scales;
+            if (DynamicResolutionHandler.instance.HardwareDynamicResIsEnabled() && m_HardwareDynamicResRequested)
+            {
+                scales = CalculateScaledRatioAgainstMaxSize(DynamicResolutionHandler.instance.finalViewport);
+            }
+            else
+            {
+                scales = CalculateRatioAgainstMaxSize(width, height);
```

Comment:
This is because the rounding is different in HW mode? Why not calling the CalculateScaledRatioAgainstMaxSize in this case too?

Reply:
So this is a bit tricky: we want to apply the scale when the target size is the final viewport, and we dont when it is not.
It doesnt make sense to apply the scale when the size is the 'smaller' drsd, since its already scaled.

Reply:
I have to figure out whats the best way to approach this

Reply:
I think the fundamental problem is that somewhere, we are dealing with scales that are not rounded to the pixel, I will do a check on the entire pipe, we might be able to remove this stupid complexity

Reply:
Ok I have a more elegant fix for this. Upcoming in the next push